spec/features/user_edits_book_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+feature 'User edits an existing book' do
+  scenario 'updates title field successfully' do
+    @book = create(:book)
+    visit root_path
+
+    edit_book(@book, new_title: 'The Golden Compass',
+                     new_author: @book.author)
+
+    expect(page.current_path).to eq books_path
+  end
+
+  scenario 'updates title and author fields successfully' do
+    @book = create(:book)
+    visit root_path
+
+    edit_book(@book, new_title: 'The Amber Spyglass',
+                     new_author: 'Philip Pullman')
+
+    expect(page.current_path).to eq books_path
+  end
+
+  def edit_book(book, options={})
```

Comment:
Advice on how to refactor this function would be great. Ideally, I want it to take a different number of arguments based on whether one field is being edited or both fields are being edited.